pyaedt/generic/ibis_reader.py

```diff
@@ -0,0 +1,739 @@
+
+
+class ModelSelectorItem:
+    def __init__(self):
+        self._description = []
+        self._name = None
+
+    @property
+    def description(self):
+        """Description of the item."""
+        return self._description
+
+    @description.setter
+    def description(self, value):
+        self._description = value
+
+    @property
+    def name(self):
+        """Name of the item."""
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        self._name = value
+
+
+class Model:
+    def __init__(self):
+        self._description = []
+        self._name = None
+        self._clamp = None
+        self._enable = None
+
+    @property
+    def name(self):
+        """Name of the item."""
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        self._name = value
+
+    @property
+    def model_type(self):
+        """Type of the model."""
+        return self._model_type
+
+    @model_type.setter
+    def model_type(self, value):
+        self._model_type = value
+
+    @property
+    def clamp(self):
+        """Clamp."""
+        return self._clamp
+
+    @clamp.setter
+    def clamp(self, value):
+        self._clamp = value
+
+    @property
+    def enable(self):
+        """Is model enabled or not."""
+        return self._enable
+
+    @enable.setter
+    def enable(self, value):
+        self._enable = value
+
+
+class Ibis:
+    """Ibis model with all data extracted: name, components, models.
+
+    Parameters
+    ----------
+    name : str
+        Name of ibis model.
+    circuit : class:`pyaedt.circuit.Circuit`
+        Circuit in which the ibis components will be used.
+    """
+
+    # Ibis reader must work independently or in Circuit.
+    def __init__(self, name, circuit):
+        self.circuit = circuit
+        self._name = name
+        self._components = {}
+        self._model_selectors = []
+        self._models = []
+
+    @property
+    def name(self):
+        """Name of the ibis model."""
+        return self._name
+
+    @property
+    def components(self):
+        """List of all components included in the ibis file."""
+        return self._components
+
+    @components.setter
+    def components(self, value):
+        self._components = value
+
+    @property
+    def model_selectors(self):
+        """List of all model selectors included in the ibis file."""
+        return self._model_selectors
+
+    @model_selectors.setter
+    def model_selectors(self, value):
+        self._model_selectors = value
+
+    @property
+    def models(self):
+        """List of all models included in the ibis file."""
+        return self._models
+
+    @models.setter
+    def models(self, value):
+        self._models = value
+
+    @property
+    def buffers(self):
+        """Buffers included into the ibis model."""
+        return self._buffers
+
+    @buffers.setter
+    def buffers(self, value):
+        self._buffers = value
+
+
+class IbisReader:
```

Comment:
There's no `__init__` defined for this class. It's unclear to me, but if `read_project` must be called for this class to be useful, I would just convert `read_project` --> `__init__`.

Reply:
The issue is that in this method `read_project` I returned an IBIS object. I would not be able to do that within the initializer.